src/components/Upload/UploadFile.vue

```diff
@@ -34,35 +56,58 @@ import axios from 'axios';
 export default {
 	data() {
 		return {
-			file: undefined,
+			dragAndDropCapable: false,
+            uploadPercentage: 0
 		};
 	},
+	mounted(){
+		this.dragAndDropCapable = this.determineDragAndDropCapable();              
+		
+		// If drag and drop capable, then we continue to bind events to our elements.
+		if ( this.dragAndDropCapable ){
+			
+			// Listen to all of the drag events and bind an event listener to each for the fileform.
+			['drag', 'dragstart', 'dragend', 'dragover', 'dragenter', 'dragleave', 'drop'].forEach( function( evt ) {
+			
+			// For each event add an event listener that prevents the default action (opening the file in the browser) 
+			// and stop the propagation of the event (so no other elements open the file in the browser)		
+			this.$refs.fileform.addEventListener(evt, function(e){
+				e.preventDefault();
+				e.stopPropagation();
+			}.bind(this), false);
+			}.bind(this));
+
+			// Add an event listener for drop to the form
+			this.$refs.fileform.addEventListener('drop', function(e){
+				//  Capture the files from the drop event and add them to our local files array.
+				for( let i = 0; i < e.dataTransfer.files.length; i++ ){
+					this.$parent.files.push( e.dataTransfer.files[i] );
```

Comment:
Why dont we use foreach here? And please, do not use the `$parent`, use emitting the event. 

Parent -> child: PROPS (https://vuejs.org/v2/guide/components.html#Passing-Data-to-Child-Components-with-Props)
Child -> Parent: EVENT EMMITING (https://vuejs.org/v2/guide/components.html#Listening-to-Child-Components-Events)